lib/posix/options/pipe.c

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) Bruce ROSIER
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "posix_features.h"
+#include "zephyr/fs/fs.h"
+#include <zephyr/posix/unistd.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/sys/fdtable.h>
+#include <sys/_intsup.h>
+
+#define NUM_PIPES        CONFIG_POSIX_NUM_PIPES
+#define PIPE_BUFFER_SIZE _POSIX_PIPE_BUF
+
+static ssize_t pipe_read_vmeth(void *obj, void *buffer, size_t count);
+static ssize_t pipe_write_vmeth(void *obj, const void *buffer, size_t count);
+static int pipe_close_vmeth(void *obj);
+static int pipe_ioctl_vmeth(void *obj, unsigned int request, va_list args);
+
+struct pipe_desc {
+	unsigned char __aligned(4) ring_buffer[_POSIX_PIPE_BUF];
```

Comment:
Zephyr has a ring-buffer implementation. It should be used here as well, I would guess.

Reply:
Hi cfriedt,

My implementation of the posix pipe is based on the ZEPHYR pipe and it needs an aligned(4) buffer to be initialized